sdks/besu/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -0,0 +1,282 @@
+import * as assetLocksPb from "@hyperledger-labs/weaver-protos-js/common/asset_locks_pb";
+import * as Web3 from 'web3-utils';
+import { Hash } from "./HashFunctions";
+
+// Create an asset exchange agreement structure
+function createAssetExchangeAgreementSerialized(assetType: string, assetID: string, locker: string, recipient: string) {
+    const assetExchangeAgreement = new assetLocksPb.AssetExchangeAgreement();
+    assetExchangeAgreement.setAssettype(assetType);
+    assetExchangeAgreement.setId(assetID);
+    assetExchangeAgreement.setRecipient(recipient);
+    assetExchangeAgreement.setLocker(locker);
+    return Buffer.from(assetExchangeAgreement.serializeBinary())
+}
+
+function createFungibleAssetExchangeAgreementSerialized(assetType: string, numUnits: number, locker: string, recipient: string) {
+    const assetExchangeAgreement = new assetLocksPb.FungibleAssetExchangeAgreement();
+    assetExchangeAgreement.setAssettype(assetType);
+    assetExchangeAgreement.setNumunits(numUnits);
+    assetExchangeAgreement.setRecipient(recipient);
+    assetExchangeAgreement.setLocker(locker);
+    return Buffer.from(assetExchangeAgreement.serializeBinary())
+}
+
+
+function createHybridAssetExchangeAgreementSerialized(assetType: string, assetID: string, assetData: string, numUnits: number, recipient: string) {
+    const assetExchangeAgreement = new assetLocksPb.HybridAssetExchangeAgreement();
+    assetExchangeAgreement.setAssettype(assetType);
+    assetExchangeAgreement.setId(assetID);
+    assetExchangeAgreement.setRecipient(recipient);
+    assetExchangeAgreement.setAssetdata(Buffer.from(assetData, "hex"));
+    assetExchangeAgreement.setNumunits(numUnits);
+    return Buffer.from(assetExchangeAgreement.serializeBinary())
+}
+
+// Create an asset lock structure
+function createAssetLockInfoSerialized(hash: Hash, expiryTimeSecs: number) {
+    const lockInfoHTLC = new assetLocksPb.AssetLockHTLC();
+    lockInfoHTLC.setHashmechanism(hash.HASH_MECHANISM);
+    lockInfoHTLC.setHashbase64(Buffer.from(hash.getSerializedHashBase64(), 'base64'));
+    lockInfoHTLC.setExpirytimesecs(expiryTimeSecs);
+    lockInfoHTLC.setTimespec(assetLocksPb.TimeSpec.EPOCH)
+    return Buffer.from(lockInfoHTLC.serializeBinary())
+}
+
+// Create an asset claim structure
+function createAssetClaimInfoSerialized(hash: Hash) {
+    const claimInfoHTLC = new assetLocksPb.AssetClaimHTLC();
+    claimInfoHTLC.setHashmechanism(hash.HASH_MECHANISM);
+    claimInfoHTLC.setHashpreimagebase64(Buffer.from(hash.getPreimage()));
+    const claimInfoHTLCSerialized = claimInfoHTLC.serializeBinary();
+    const claimInfo = new assetLocksPb.AssetClaim();
+    claimInfo.setLockmechanism(assetLocksPb.LockMechanism.HTLC);
+    claimInfo.setClaiminfo(claimInfoHTLCSerialized);
+    return Buffer.from(claimInfo.serializeBinary())
+}
+
+const createHTLC = async (
+    assetManagerContract: any,
+    tokenContract: any,
+    assetType: string,
+    assetID: string,
+    senderAddress: string,
```

Comment:
senderAddress -> lockerAddress

Reply:
@sanvenDev  why change the name to lockeraddress? I thought we still need to pass the sender address since the transaction is initiate from sender addresss

Reply:
@michiboo sender and locker are same entity right? and just to be consistent, locker and recipient makes more sense in HTLC.

Reply:
@michiboo We want to be consistent in our terminology. "Locker" is what we have been using in other Weaver modules, because the proximate purpose of the operation is the locking of an asset. "Sending" is a goal that is fulfilled by multiple HTLC operations.